src/test/java/HostServerTest.java

```diff
@@ -0,0 +1,56 @@
+import echoserver.ClientConnection;
+import echoserver.HostServer;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+
+class FakeServerSocket extends ServerSocket{
```

Comment:
Can you talk about why you chose to extend ServerSocket instead of introducing an interface?